### DISCUSSION TITLE
Calling Existing Macros From the API

### DISCUSSION THREAD

**Discussion by btcCharlie** [chhoumann/quickadd#311] [Q&A] [answered]
Hi! I make heavy use of user scripts with `quickadd`, and I'd love to know if there's a hidden API functionality for calling arbitrary quickadd macros from any user script. It would allow me to streamline my note creation process even further.

I know there's the format syntax for firing up macros `{{MACRO:<macro>}}`, so I suspect there's a hidden functionality for this, which isn't exposed in the API or maybe even just isn't documented. Although I don't want to assume facilitating this has to be easy.

Is there something like this I can reference immediately? Or is calling macros from within other macros exclusively a format syntax thing?

---

> Some specifics to my issue: I'm trying to build myself a project/task management vault with timesheeting, and I've run into a situation where I would need to call a macro I made from within a user script in another macro. Specifically, when creating a timesheet, it's supposed to link to a sprint. As of now, iif no suitable (active, date range) sprint is found, it fails to create the timesheet. 
>
> Ideally, I'd like to show a prompt asking whether the user wants to create a new sprint as well, and fire up a macro for creating it, having an immediate reference in the calling user script to link to after its completion.

**Comment by holroy** [comment, marked as answer]
I've been looking for something similar, but I'm in the start phase of utilising Quick Add, so there might be a better way, but could [executeChoice()](https://quickadd.obsidian.guide/docs/QuickAddAPI#executechoicechoicename-string-variables-key-string-any-promise) be the thing you're looking for?

If you add your macro to the list of choices, you should then be able to call it using this function, if I understand the documentation correctly.
